File: Trivia_Alon_Auri/Requests.cpp
#include "Requests.h"
#include <cstdint>
#include <iostream>
#include <vector>

using nlohmann::json;

void from_json(const json& j, LoginRequest& request) {
	j.at("Username").get_to(request.username);
	j.at("Password").get_to(request.password);
}

void from_json(const json& j, SignupRequest& request) {
	j.at("Username").get_to(request.username);
	j.at("Password").get_to(request.password);
	j.at("Email").get_to(request.email);
	j.at("Address").get_to(request.address);
	j.at("PhoneNumber").get_to(request.phoneNumber);
	j.at("BirthDate").get_to(request.birthDate);
}

void from_json(const nlohmann::json& j, CreateRoomRequest& request)
{
	j.at("RoomName").get_to(request.roomName);
	j.at("MaxUsers").get_to(request.maxUsers);
	j.at("QuestionCount").get_to(request.questionCount);
	j.at("AnswerTimeout").get_to(request.answerTimeout);
}

void from_json(const nlohmann::json& j, GetPlayersInRoomRequest& request)
{
	j.at("RoomId").get_to(request.roomId);
}

void from_json(const nlohmann::json& j, JoinRoomRequest& request)
{
	j.at("RoomId").get_to(request.roomId);
}

File: Trivia_Client/Requests.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Trivia_Client
{
    class Requests
    {
        public class Request
        {
        }

        public class LoginRequest : Request
        {
            public String Username { get; set; }
            public String Password { get; set; }
        }

        public class SignupRequest : Request
        {
            public String Username { get; set; }
            public String Password { get; set; }
            public String Email { get; set; }
            public String Address { get; set; }
            public String PhoneNumber { get; set; }
            public String BirthDate { get; set; }
        }

        public class GetPlayersInRoomRequest : Request
        {
            public int RoomId { get; set; }
        }

        public class JoinRoomRequest : Request
        {
            public int RoomId { get; set; }
        }

        public class CreateRoomRequest : Request
        {
            public String RoomName { get; set; }
            public int MaxUsers { get; set; }
            public int QuestionCount { get; set; }
            public int AnswerTimeout { get; set; }
        }

    }
}
